.github/workflows/on-push-main.yml

```diff
@@ -2,9 +2,7 @@ name: Build
 on:
   push:
     branches:
-      - '*'
-    tags-ignore:
-      - '*'
+      - main
```

Comment:
It was better to split this out instead of trying to have `on-push.yml` handle extra jobs when only pushing to a PR since there was issues determining if it was a PR or not, but that has led to more build code being duplicated across `on-push-main.yml`, `on-tag.yml`, and `on-pull-request.yml`. I am not as familiar with GitHub Actions as I am GitLab CI so I'm not sure if there's an intuitive way to dedupe code. This would be a nice gold plating opportunity in the future

Small rant: Why are popular CI's all yml based? I would love for CI code I write to be statically typed, instead of the constant commit, wait for CI, then see if it worked. Such a time sink